references/embedding/loss.py

```diff
@@ -0,0 +1,113 @@
+'''
+    Pytorch adaptation of https://omoindrot.github.io/triplet-loss
+    https://github.com/omoindrot/tensorflow-triplet-loss
+'''
+import torch
+from torch.nn.modules.loss import _Loss
+from torch._jit_internal import weak_module, weak_script_method
+
+
+@weak_module
```

Comment:
Can we remove the `weak_module` stuff? This is going away in PyTorch. You don't need to inherit from `_Loss`, which is an internal thing in PyTorch. Also, you don't use the `reduction` / `reduce` / `size_average` anywhere in your `forward`, so those can be all removed and just inherit from `nn.Module`.

Reply:
Done.